src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,37 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_USER_AGE = 18;
+    private static final int MIN_PASS_LENGTH = 6;
+    private final StorageDao storageDao;
+
+    public RegistrationServiceImpl() {
+        storageDao = new StorageDaoImpl();
+    }
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RuntimeException("User can't be null");
+        }
+        if (user.getLogin() == null) {
+            throw new RuntimeException("Login can't be null");
+        }
+        if (user.getLogin().isEmpty()) {
+            throw new RuntimeException("Login can't be empty");
+        }
+        if (user.getPassword() == null) {
+            throw new RuntimeException("Password can't be null");
+        }
+        if (user.getAge() == null) {
+            throw new RuntimeException("Age can't be null");
+        }
+        if (user.getAge() < MIN_USER_AGE) {
+            throw new RuntimeException("You can't add the user with age is less 18 years old");
+        }
+        if (user.getPassword().length() < MIN_PASS_LENGTH) {
+            throw new RuntimeException("You can't add the user with password is less 6 characters");
+        }
+        return storageDao.get(user.getLogin()) == null ? storageDao.add(user) : null;
```

Comment:
```suggestion
        if (storageDao.get(user.getLogin()) != null) {
            throw new RuntimeException("User by login " + user.getLogin() + " already exists");
        }
        return storageDao.add(user);
```
it is registration service so if login is taken it is also an exception